pkg/api/vlabs/upgradetypesandvalidate.go

```diff
@@ -0,0 +1,27 @@
+package vlabs
+
+import "fmt"
+
+// UpgradeContainerService API model
+type UpgradeContainerService struct {
+	OrchestratorProfile *OrchestratorProfile `json:"orchestratorProfile,omitempty"`
+}
+
+// Validate implements APIObject
+func (ucs *UpgradeContainerService) Validate() error {
```

Comment:
That's the validation I was referring to earlier that I still need to do. Though I don't think this is the right place for it.